src/ray/raylet/worker_pool.h

```diff
@@ -535,6 +569,26 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
   /// worker types (SPILL_WORKER and RESTORE_WORKER and UTIL_WORKER).
   bool IsIOWorkerType(const rpc::WorkerType &worker_type);
 
+  /// Call the `PopWorkerCallback` function asynchronously to make sure executed in
+  /// different stack.
+  virtual void PopWorkerCallbackAsync(const PopWorkerCallback &callback,
+                                      std::shared_ptr<WorkerInterface> worker,
+                                      PopWorkerStatus status = PopWorkerStatus::OK);
+
+  /// Try to assign task which waiting for workers.
+  /// \param workers_to_tasks The queue of tasks which waiting for workers.
+  /// \param proc The process which the worker belongs to.
+  /// \param worker A new idle worker. If the worker is empty, we could also callback to
+  /// the task. \param status The pop worker status which will be forwarded to
+  /// `PopWorkerCallback`. \param found  Whether the related task found or not. \param
+  /// dispatched Whether the related task dispatched or not. \param task_id  The related
+  /// task id.
```

Comment:
better rename `dispatched` to `worker_used`: "Whether the worker is used by the task, only valid when found is true."